src/Surfnet/ServiceProviderDashboard/Application/CommandHandler/Entity/PublishEntityProductionCommandHandler.php

```diff
@@ -48,48 +42,50 @@ class PublishEntityProductionCommandHandler implements CommandHandler
      * @var PublishEntityRepository
      */
     private $publishClient;
-
     /**
-     * @var CommandBus
+     * @var TicketService
      */
-    private $commandBus;
+    private $ticketService;
 
     /**
-     * @var MailMessageFactory
+     * @var FlashBagInterface
      */
-    private $mailMessageFactory;
+    private $flashBag;
 
     /**
-     * @var TokenStorageInterface
+     * @var LoggerInterface
      */
-    private $tokenStorage;
+    private $logger;
 
     /**
-     * @var FlashBagInterface
+     * @var string
      */
-    private $flashBag;
+    private $issueType;
 
     /**
-     * @var LoggerInterface
+     * PublishEntityProductionCommandHandler constructor.
+     * @param EntityRepository $entityRepository
+     * @param PublishEntityRepository $publishClient
+     * @param TicketService $ticketService
+     * @param FlashBagInterface $flashBag
+     * @param LoggerInterface $logger
+     * @param string $issueType
      */
-    private $logger;
-
     public function __construct(
         EntityRepository $entityRepository,
         PublishEntityRepository $publishClient,
-        CommandBus $commandBus,
-        MailMessageFactory $mailMessageFactory,
-        TokenStorageInterface $tokenStorage,
+        TicketService $ticketService,
         FlashBagInterface $flashBag,
-        LoggerInterface $logger
+        LoggerInterface $logger,
+        $issueType
```

Comment:
I think it would be better if this parameter is mandatory to prevent a lot of if statements later on. Or perhaps remove it here if it's not the responsibility of the ticket.

Reply:
The parameter is required, but you might mean the parameters.yml parameter? That too is an application requirement. I think bootstrapping will crash if the parameter is not set, the assertion is a mere DiD & clarification means to ensure you know where that parameter should come from.

Reply:
Sorry, I meant the Ticket::issueType